### PR TITLE
feat: Add add autogenerated javadoc sample for selecting REST transport over gRPC

### PR DESCRIPTION
**Pull request by vam-google**
This is for `grpc+rest` clients.

Also add proper `grpc+rest` unit tests which were missing.

### CODE REVIEW

**Comment by vam-google** [comment]
@blakeli0 PTAL

File: src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java [hunk 1]
```diff
@@ -112,7 +114,10 @@ public static List<CommentStatement> createClassHeaderComments(
       Service service,
       String classMethodSampleCode,
       String credentialsSampleCode,
-      String endpointSampleCode) {
+      String endpointSampleCode,
+      String transportStample,
```
Comment (Neenu1995) [inline src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java:118]: typo?

Comment (vam-google) [reply src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java:118]: Yep. Fixed

File: src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceClientClassComposer.java [hunk 2]
```diff
@@ -208,7 +208,10 @@ private static List<CommentStatement> createClassHeaderComments(
         service,
         SampleCodeWriter.writeInlineSample(classMethodSampleCode.body()),
         SampleCodeWriter.writeInlineSample(credentialsSampleCode.body()),
-        SampleCodeWriter.writeInlineSample(endpointSampleCode.body()));
+        SampleCodeWriter.writeInlineSample(endpointSampleCode.body()),
```
Comment (eaball35) [inline src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceClientClassComposer.java:211]: do you mean to include these null args?

File: src/main/java/com/google/api/generator/gapic/composer/grpcrest/ServiceClientClassComposer.java [hunk 3]
```diff
@@ -26,4 +40,37 @@ protected ServiceClientClassComposer() {
...
+    Sample transportSampleCode =
+        ServiceClientHeaderSampleComposer.composeTransportSample(
+            clientType, settingsType, "defaultHttpJsonTransportProviderBuilder");
+    samples.addAll(
+        Arrays.asList(
+            classMethodSampleCode, credentialsSampleCode, endpointSampleCode, transportSampleCode));
+
+    return ServiceClientCommentComposer.createClassHeaderComments(
+        service,
+        SampleCodeWriter.writeInlineSample(classMethodSampleCode.body()),
```
Comment (eaball35) [inline src/main/java/com/google/api/generator/gapic/composer/grpcrest/ServiceClientClassComposer.java:69]: You could also combine all the different sample.body into one List<Statement> and pass the whole list to SampleCodeWriter.writeInlineSample()

File: src/test/java/com/google/api/generator/gapic/composer/grpcrest/GrpcServiceCallableFactoryClassComposerTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    Service echoProtoService = context.services().get(0);
+    GapicClass clazz =
+        GrpcServiceCallableFactoryClassComposer.instance().generate(context, echoProtoService);
+
+    JavaWriterVisitor visitor = new JavaWriterVisitor();
+    clazz.classDefinition().accept(visitor);
+    Utils.saveCodegenToFile(this.getClass(), "GrpcEchoCallableFactory.golden", visitor.write());
+    Path goldenFilePath =
+        Paths.get(Utils.getGoldenDir(this.getClass()), "GrpcEchoCallableFactory.golden");
+    Assert.assertCodeEquals(goldenFilePath, visitor.write());
```
Comment (eaball35) [inline src/test/java/com/google/api/generator/gapic/composer/grpcrest/GrpcServiceCallableFactoryClassComposerTest.java:41]: Consider checking the clazz.samples() as well. See similar [grpc tests](https://github.com/googleapis/gapic-generator-java/blob/main/src/test/java/com/google/api/generator/gapic/composer/grpc/ServiceClientClassComposerTest.java#L52).  See [Assert.assertGoldenSamples](https://github.com/googleapis/gapic-generator-java/blob/051713dec7706713900b8708e3a6b9ab0639438e/src/test/java/com/google/api/generator/test/framework/Assert.java#L68) and [assertEmptySamples](https://github.com/googleapis/gapic-generator-java/blob/051713dec7706713900b8708e3a6b9ab0639438e/src/test/java/com/google/api/generator/test/framework/Assert.java#L53). Even if there aren't any samples, I think it's good to be clear in the tests which classes generate samples and which don't.

**Pull request merged**